docs/src/main/znai/auth/auth-and-impersonation.md

```diff
@@ -0,0 +1,79 @@
+Liftwizard supports the dynamic configuration of [dropwizard-auth](https://www.dropwizard.io/en/stable/manual/auth.html) in `config.yml` and the configuration of impersonation in tests.
+
+Add a dependency on `liftwizard-bundle-auth-filter` and add `AuthFilterBundle` to the registered bundles.
+
+Modify the application's configuration class to implement `AuthFilterFactoryProvider` if it does not already extend `AbstractLiftwizardConfiguration`.
+
+## Test configuration
+
+For tests, you'll typically want to use impersonation for authentication based on headers.
+
+Add a dependency on `liftwizard-config-auth-filter-header`.
+
+```xml
+<dependency>
+    <groupId>io.liftwizard</groupId>
+    <artifactId>liftwizard-config-auth-filter-header</artifactId>
+    <scope>test</scope>
+</dependency>
+```
+
+Add an `authFilters` list to `config-test.json5` containing just one filter, of type `header`.
+
+```json5
+{
+  authFilters: [
+    {
+      type: "header",
+      header: "Authorization",
+      prefix: "Impersonation",
+    },
+  ],
+}
+```
+
+## Test code
+
+Impersonation authorization can be used from any test that uses Dropwizard test utilities, like `DropwizardAppExtension`, `LiftwizardAppExtension`, or `DropwizardAppRule`.
+
+```java
+@Test
+void smokeTest()
+{
+    Client client = this.appExtension.client();
+
+    Response response = client
+            .target("http://localhost:{port}/api/example")
+            .resolveTemplate("port", this.appExtension.getLocalPort())
+            .request()
+            .header("Authorization", "Impersonation User ID")
+            .get();
+
+    // add assertions here
+}
+```
+
+Whenever we use `dropwizard-auth`, some of our Jersey resource methods will be authenticated. The authenticated methods will be annotated with a security annotation such as `@PermitAll`. The user principal will be passed in as a parameter and annotated like `@Auth Principal principal`. The string passed in the header (`"Impersonation User ID"` in this example) will be accessible via `principal.getName()`.
```

Comment:
Just User ID, not Impersonation User ID, right? The prefix would get chopped off?

Reply:
Good catch, thanks.